wp-content/themes/core/components/blocks/icon_grid/Icon_Grid_Controller.php

```diff
@@ -0,0 +1,251 @@
+			],
+		];
+	}
+
+	/**
+	 * @return Deferred_Component
+	 */
+	private function get_leadin(): Deferred_Component {
+		return defer_template_part( 'components/text/text', null, [
+			Text_Controller::CLASSES => [
+				'c-block__leadin',
+				'b-icon-grid__leadin',
+				'h6',
+			],
+			Text_Controller::CONTENT => $this->leadin ?? '',
+		] );
+	}
+
+	/**
+	 * @return Deferred_Component
+	 */
+	private function get_title(): Deferred_Component {
+		return defer_template_part( 'components/text/text', null, [
+			Text_Controller::TAG     => 'h2',
+			Text_Controller::CLASSES => [
+				'c-block__title',
+				'h3',
+			],
+			Text_Controller::CONTENT => $this->title ?? '',
+		] );
+	}
+
+	/**
+	 * @return Deferred_Component
+	 */
+	private function get_content(): Deferred_Component {
+		return defer_template_part( 'components/container/container', null, [
+			Container_Controller::CLASSES => [
+				'c-block__description',
+				'b-icon-grid__description',
+				't-sink',
+				's-sink',
+			],
+			Container_Controller::CONTENT => $this->description ?? '',
+		] );
+	}
+
+	/**
+	 * @return Deferred_Component
+	 */
+	public function get_cta(): Deferred_Component {
+		$cta = wp_parse_args( $this->cta, [
+			'content' => '',
+			'url'     => '',
+			'target'  => '',
+		] );
+
+		return defer_template_part( 'components/link/link', null, [
+			Link_Controller::URL     => $cta['url'],
+			Link_Controller::CONTENT => $cta['content'] ?: $cta['url'],
+			Link_Controller::TARGET  => $cta['target'],
+			Link_Controller::CLASSES => [
+				'c-block__cta-link',
+				'a-btn',
+				'a-btn--has-icon-after', 
+				'icon-arrow-right'
+			],
+		] );
+	}
+
+	public function get_icon_card_args() {
+		if ( empty( $this->icons ) ) {
+			return [];
+		}
+
+		$cards = [];
+		foreach ( $this->icons as $card ) {
+			$cards[] = [
+				Card_Controller::STYLE           => Card_Controller::STYLE_PLAIN,
+				Card_Controller::TAG             => 'li',
+				Card_Controller::CLASSES         => ['is-centered-text'],
+				Card_Controller::USE_TARGET_LINK => false,
+				Card_Controller::TITLE           => defer_template_part(
+					'components/text/text',
+					null,
+					[
+						Text_Controller::TAG     => 'h3',
+						Text_Controller::CLASSES => [ 'h5' ],
+						Text_Controller::CONTENT => $card['icon_title'] ?? '',
+					]
+				),
+				Card_Controller::DESCRIPTION     => defer_template_part(
+					'components/container/container',
+					null,
+					[
+						Container_Controller::CONTENT => wpautop( $card['icon_description'] ) ?? '',
+						Container_Controller::CLASSES => [ 't-sink', 's-sink' ],
+					],
+				),
+				Card_Controller::IMAGE           => defer_template_part(
+					'components/image/image',
+					null,
+					[
+						Image_Controller::IMG_ID       => $card['image'] ?? null,
+						Image_Controller::AS_BG        => false,
+						Image_Controller::SRC_SIZE     => 'medium_large',
+						Image_Controller::SRCSET_SIZES => [
+							'medium',
+							'medium_large',
+						],
```

Comment:
Same question here re: these image sizes. Are these actual sizes in WordPress?

Reply:
It's pretty hard to find the documentation for it but it's listed here as an option: https://codex.wordpress.org/Post_Thumbnails#Thumbnail_Sizes